src/nyc_trees/apps/home/training/types.py

```diff
@@ -199,25 +199,66 @@ def extract_answers(cls, post_fields):
     def score(self, answers):
         """
         How many correct answers are selected?
-        answers - dict of {question_index => answer_index, ...}
+        answers - dict of {question_index => answer_indexes, ...}
         """
         result = 0
         for i, question in enumerate(self.questions):
-            if answers.get(i, -1) == question.answer:
+            answer = answers.get(i, None)
+            if question.is_correct(answer):
                 result += 1
         return result
 
 
-class Question(object):
+class AbstractQuestion(object):
     """
+    Base class for Quiz questions.
     text - string
-    answer - int; Index of the correct answer in choices iterable
     choices - iterable<string>
     """
-    def __init__(self, text, answer, choices):
+    def __init__(self, text, choices):
         assert len(choices) > 0
+        self.text = unicode(text)
+        self.choices = list(choices)
+
+    def is_correct(self, answer):
+        """
+        answer - iterable<int> or None
+        """
+        raise NotImplementedError()
+
+    def is_multiple_choice(self):
+        return False
+
+
+class Question(AbstractQuestion):
+    """
+    Represents a question with only one correct answer.
+    answer - int; Index of the correct answer in choices iterable
+    """
+    def __init__(self, text, choices, answer):
+        super(Question, self).__init__(text, choices)
         assert answer >= 0
         assert answer < len(choices)
-        self.text = unicode(text)
         self.answer = int(answer)
-        self.choices = list(choices)
+
+    def is_correct(self, answer):
+        return answer and len(answer) > 0 and answer[0] == self.answer
+
+
+class MultiChoiceQuestion(AbstractQuestion):
+    """
+    Represents a question with multiple correct answers.
+    answer - iterable<int>; Indexes of the correct answers in choices iterable
+    """
+    def __init__(self, text, choices, answer):
+        super(MultiChoiceQuestion, self).__init__(text, choices)
+        for ans in answer:
+            assert ans >= 0
+            assert ans < len(choices)
+        self.answer = set(answer)
+
+    def is_correct(self, answer):
+        return answer and set(answer) == self.answer
```

Comment:
Good observation. The idea is that `set` is an implementation detail. As long as you define the quiz answer as some iterable (tuple, array, generator, etc.) it will work.